tests/measurements/test_measurements.py

```diff
@@ -648,6 +648,25 @@ def circuit():
         ):
             circuit()
 
+    def test_state_measurement_process_density_matrix_not_implemented(self):
+        """Test that the process_density_matrix method of StateMeasurement raises
+        NotImplementedError."""
+
+        class MyMeasurement(StateMeasurement):
+            def process_state(self, state, wire_order):
+                return qml.math.sum(state)
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            return MyMeasurement().process_density_matrix(
+                density_matrix=qml.math.array([[1, 0], [0, 0]]), wire_order=Wires([0, 1])
+            )
+
+        with pytest.raises(NotImplementedError):
+            circuit()
```

Comment:
There is no need to put this inside a `QNode` as you can directly test that calling `MyMeasurement().process_density_matrix` raises an error.

Reply:
Cool, let me simplify this soon later